modules/smitten/index.ts

```diff
@@ -10,10 +10,16 @@ export interface Smitten<T extends string | number = string | number> {
 }
 
 class DefaultEmitter implements Smitten {
+  private cache: {
+    [key: string]: DefaultEmitter
+  }
+
   constructor(
     readonly type: string | number,
     readonly parent: DefaultEmitter | RootEmitter
-  ) {}
+  ) {
+    this.cache = {}
```

Comment:
This can be moved up with the declaration [here](https://github.com/action-land/action-land/pull/21/files#diff-5f9ebf78bafb322f2ca85b3c2a8ba4bdR14).